cpp/src/arrow/compute/exec/asof_join_node_test.cc

```diff
@@ -662,19 +662,19 @@ TRACED_TEST_P(AsofJoinBasicTest, TestBasic1, {
   runner(basic_test);
 })
 
-BasicTest GetBasicTest1Negative() {
```

Comment:
Let's name these tests "backward" or "forward" similar to how pandas call them:

https://pandas.pydata.org/pandas-docs/version/0.25.0/reference/api/pandas.merge_asof.html#pandas-merge-asof